onnxruntime/core/framework/prepacked_weights_container.h

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <unordered_map>
+#include <unordered_set>
+#include <string>
+#include <cstdint>
+
+#include "core/framework/buffer_deleter.h"
+
+#include "core/framework/allocator.h"
+#include "core/platform/ort_mutex.h"
+#include "prepacked_weights.h"
+
+namespace onnxruntime {
+
+class PrepackedWeightsContainer final {
+ public:
+  PrepackedWeightsContainer() {
+  }
+
+  ~PrepackedWeightsContainer() = default;
+
+  AllocatorPtr GetAllocator(const std::string& device_name);
+
+  const PrePackedWeights& GetCachedWeight(const std::string& key);
+
+  void WriteCachedWeight(const std::string& key, PrePackedWeights&& packed_weight);
+
+  bool HasCachedWeight(const std::string& key) const;
+
+  bool HasPrepackedWeightForOpTypeAndConstantInitializer(const std::string& op_type,
+                                                         const void* const_initialized_tensor_data) const;
+
+  void MarkHasPrepackedWeightForOpTypeAndConstantInitializer(const std::string& op_type,
```

Comment:
Thinking more about this - do we really need this? We already know that weights can be pre-packed differently across different instances of the same op. What additional value is this serving? We can simplify the code by removing this.

Reply:
removed this altogether